lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -586,4 +601,33 @@ defmodule RealtimeWeb.RealtimeChannel do
     Logger.error("Start channel error: " <> error_msg)
     {:error, %{reason: error_msg}}
   end
+
+  defp assign_policies(false, db_conn, sub_topic, access_token, claims, socket) do
+    channel =
+      case ChannelsCache.get_channel_by_name(sub_topic, db_conn) do
+        {:error, :not_found} -> raise "Channel #{sub_topic} hasn't been initialized"
+        {:ok, channel} -> channel
+      end
+
+    authorization_context =
+      Authorization.build_authorization_params(%{
+        channel: channel,
+        headers: socket.assigns.headers,
+        jwt: access_token,
+        claims: claims,
+        role: claims["role"]
+      })
+
+    {:ok, socket} = Authorization.get_authorizations(socket, db_conn, authorization_context)
```

Comment:
Ah ... if you set the authorization context when you query for the Channel, if RLS says you can't read the Channel than it won't get returned in the query. 

So you're hitting the db unnecessariy above really.

Reply:
We'll need to cache this too. 

IMPORTANT: ttl on the specific key needs to be set to the same time the JWT expires

Reply:
tricky thing here is that we need to fetch the channel before the config is applied

we might be able to do it differently:
* Use only channel_name on the config as is
* After channel read permissions are checked, fetch the channel so we have access to the channel.id which is needed to fetch the broadcast information

we could also modify broadcast to also include channel_name and then we would not need the channel

Reply:
wdyt?

Reply:
Seem okay tho, and maybe a good way to error on "create this channel first" so you know it's created vs created but filtered because of RLS rules.